packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts

```diff
@@ -0,0 +1,291 @@
+import { FunctionParameters } from 'amplify-function-plugin-interface';
+import { SUPPORTED_SERVICES } from '../supported-services';
+import { serviceName, provider } from './utils/constants';
+import { copyTemplateFiles, createParametersFile } from './utils/copyFiles';
+import { MergeableFunctionParameters } from './MergeableFunctionParameters';
+
+const fs = require('fs-extra');
+const path = require('path');
+const inquirer = require('inquirer');
+const { invokeFunction } = require('./utils/invoke');
+
+let serviceMetadata;
+
+/**
+ * Entry point for creating a new function
+ * @param context Amplify Core Context object
+ * @param category The resource category (should always be 'function')
+ * @param service The cloud service that is providing the category
+ * @param options Legacy parameter
+ * @param parameters Parameters used to define the function. If not specified, a walkthrough will be launched to populate it.
+ */
+export async function addResource(context, category, service, options, parameters?: FunctionParameters): Promise<string> {
+  // load the service config for this service
+  const serviceConfig = SUPPORTED_SERVICES[service];
+  if (!serviceConfig) {
+    throw `amplify-category-function is not configured to provide service type ${service}`;
+  }
+  // initialize the template parameters
+  const funcParams = new MergeableFunctionParameters({
+    provider: provider,
+    service: serviceName,
+    projectName: context.amplify.getProjectDetails().projectConfig.projectName,
+  });
+
+  // merge in given parameters
+  funcParams.merge(parameters);
```

Comment:
The use case for this is currently the REST API flow supplies the user-defined path to the function add method when creating the lambda that will back the API. I agree that there should be some guardrails in place but I'm not sure what they would be at this point and since this use case is not part of our documented plugin functionality I think we can punt on it for now.